components/reporters/vulnerability-logger/internal/reporter/reporter.go

```diff
@@ -0,0 +1,233 @@
+
+// Report logs the findings in json format.
+func (j vulnReporter) Report(
+	ctx context.Context,
+	findings []*vf.VulnerabilityFinding,
+) error {
+
+	for _, finding := range findings {
+		fields, err := j.getWhatWeCareAbout(ctx, finding)
+		if err != nil {
+			return errors.Errorf("could not extract what we care about: %w", err)
+		}
+		if j.cfg.tableFormat {
+			if err := j.formatTable(ctx, fields); err != nil {
+				return errors.Errorf("could not format table: %w", err)
+			}
+			// If table format is enabled, we skip the JSON output.
+			continue
+		}
+		b, err := json.Marshal(&fields)
+		if err != nil {
+			return errors.Errorf("could not json marshal cV: %w", err)
+		}
+		fmt.Println(string(b))
+	}
+
+	return nil
+}
+
+func (j vulnReporter) formatTable(_ context.Context, finding *smithyFinding) error {
+	out, err := json.Marshal(finding)
+	if err != nil {
+		return errors.Errorf("could not json marshal finding: %w", err)
+	}
+	var data map[string]interface{}
+	if err := json.Unmarshal(out, &data); err != nil {
+		return errors.Errorf("could not unmarshal json output: %w", err)
+	}
+
+	// Collect all keys (headers)
+	keySet := map[string]struct{}{}
+	for k := range data {
+		keySet[k] = struct{}{}
+	}
+
+	// Sort keys for consistent column order
+	keys := make([]string, 0, len(keySet))
+	for k := range keySet {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	// Prepare table
+	table := tablewriter.NewWriter(os.Stdout)
+	table.Header(keys)
+
+	for _, rowData := range data {
+		table.Append(rowData)
+	}
+
+	table.Render()
+	return nil
+}
+
+func (j vulnReporter) getWhatWeCareAbout(_ context.Context, finding *vf.VulnerabilityFinding) (*smithyFinding, error) {
+	cV := smithyFinding{}
+	cV.Confidence = finding.Finding.GetConfidenceId().String()
+	cV.Enrichments = make([]*vEnrichment, 0)
+	for _, e := range finding.Finding.GetEnrichments() {
+		cV.Enrichments = append(cV.Enrichments, &vEnrichment{
+			Provider:       e.GetProvider(),
+			Value:          e.GetValue(),
+			EnrichmentName: e.GetName(),
+			Type:           e.GetType(),
+		})
+	}
+	cV.Title = finding.Finding.GetFindingInfo().GetTitle()
+	ds := finding.Finding.GetFindingInfo().GetDataSources()
+	for _, datasource := range ds {
+		datas := ocsffindinginfo.DataSource{}
+		if err := protojson.Unmarshal([]byte(datasource), &datas); err != nil {
+			return nil, errors.Errorf("could not json unmarshal finding: %w", err)
+		}
+		cV.Datasources = append(cV.Datasources, &vDatasource{
+			TargetType: datas.GetTargetType().String(),
+			Uri:        datas.GetUri().GetPath(),
+			RepoURL:    datas.GetSourceCodeMetadata().GetRepositoryUrl(),
+			Reference:  datas.GetSourceCodeMetadata().GetReference(),
+			Purl:       datas.GetOciPackageMetadata().GetPackageUrl(),
+			Tag:        datas.GetOciPackageMetadata().GetTag(),
+		})
+	}
+	cV.Description = finding.Finding.GetMessage()
+	cV.Severity = finding.Finding.GetSeverityId().String()
+	cV.Code = make([]vCode, 0)
+	cV.Pkg = make([]vPkg, 0)
```

Comment:
We seem to be missing 'findingInfo'->>'uid' as the tool provided type of finding (Finding Type in the frontend) and also created_at